openff/toolkit/topology/molecule.py

```diff
@@ -117,8 +117,8 @@ def molecule(self, molecule):
         Set the particle's molecule pointer. Note that this will only work if the particle currently
         doesn't have a molecule
         """
-        # TODO: Add informative exception here
-        assert self._molecule is None
+        err = f"{type(self).__name__} already has an associated molecule"
+        assert self._molecule is None, err
```

Comment:
Nit: This is an improvement (at least this `AssertionEror` now provides informative context) but I think the goal is to have both a custom message and exception i.e. `MoleculeAlreadyAssociatedWithSomethingError` #771

Reply:
In that case I will remove it from this PR. That is a very specific error type, though. Does it need its own type?

Reply:
IMO the change can stay  (it's an improvement) but the TODO should remain as part of the pending exception rework moving away from built-in excepitons.

I think the specificity of each exception is somewhat a matter of taste and we haven't converged on how specific or general we want exceptions to be. Something more general that covers a few different errors would probably be better. I now see these lines are in the `Particle` class whereas I previously assumed this was a `TopologyMolecule`/`ReferenceMolecule` bookkeeping error, so that exception doesn't make sense anyway.